storage/azure/src/test/integration/backend/AzureServerStorage.test.ts

```diff
@@ -0,0 +1,112 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import "reflect-metadata";
+
+import { BlobClient, ContainerClient } from "@azure/storage-blob";
+
+import {
+  buildObjectKey,
+  buildObjectReference,
+} from "@itwin/object-storage-core/lib/common/internal";
+
+import { ObjectReference, TransferConfig } from "@itwin/object-storage-core";
+import {
+  InternalStorageIntegrationTests,
+  TestRemoteDirectory,
+  TestRemoteDirectoryManager,
+  testDeleteObjectWithTransferConfig,
+  testListObjectsWithTransferConfig,
+} from "@itwin/object-storage-tests-backend";
+
+import {
+  assertAzureTransferConfig,
+  buildBlobUrl,
+} from "../../../common/internal";
+import { AzureServerStorageBindings } from "../../../server";
+import { ServerStorageConfigProvider } from "../ServerStorageConfigProvider";
+
+describe("AzureServerStorage internal tests", () => {
+  const serverStorageConfig = new ServerStorageConfigProvider().get();
+  const config = {
+    ServerStorage: {
+      dependencyName: "azure",
+      instanceName: "primary",
+      ...serverStorageConfig,
+    },
+  };
+
+  const testInfrastructure = new InternalStorageIntegrationTests(
+    config,
+    AzureServerStorageBindings
+  );
```

Comment:
This to me looks very complicated. We have very similar setup for those common tests but `StorageIntegrationTests` class does a bit more - it binds the classes, resolves storages but it also runs tests defined in that package. Those tests do not have dependencies on specific storage packages so they cannot construct tested classes directly. These new storage specific tests however can reference the tested classes directly and construct them as is - `new AzureServerStorage(new ...)`. The main drawback I see with this is that the `inversify` method is the preferred one to get storage instances, I do however think that it really complicates things in this scenario and makes the tests hard to read.

Reply:
Updated. Not sure if you could call the new way easier but it also solves the problem with using the same `Config` file as common tests (because this config is used by various helper functions and classes). Now storage specific and common tests should be properly seperated, so I believe the new way is better.

Reply:
I did not initially think it would take that much code to construct those storage instances but I agree, the end result is easier to understand.